src/main/java/com/google/devtools/build/lib/skyframe/toolchains/SingleToolchainResolutionFunction.java

```diff
@@ -284,18 +293,41 @@ private static SingleToolchainResolutionValue resolveConstraints(
     return SingleToolchainResolutionValue.create(toolchainTypeInfo, resolvedToolchainLabels);
   }
 
+  /**
+   * Helper enum to define the three indentation levels used in {@link debugMessage}.
+   */
+  private static enum IndentLevel  {
+      TargetPlatformLevel(""),
+      ToolchainLevel("  "),
+      ExecutionPlatformLevel("    ");
+
+      final String value;
+
+      IndentLevel(String value) {
+          this.value = value;
+      }
+
+      @Override
+      public String toString() {
```

Comment:
Don't override `toString` for this: add a new `public String indent()` method and call that instead.